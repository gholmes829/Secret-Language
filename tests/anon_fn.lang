# will anonomous fns work?

void call(([num] -> num) cb, num arg) {
    print(cb(arg))
}

call((num arg) -> num {
    print('doubling!')
    return arg * 2  # double trouble
}, 5)