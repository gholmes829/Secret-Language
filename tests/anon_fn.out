doubling!
10.0
